src/FilesystemSyncTaskHelper.php

```diff
@@ -56,20 +57,51 @@ class FilesystemSyncTaskHelper
      */
     private $delete_broken;
 
+    /**
+     * @var FlysystemAssetStore
+     */
+    private $flysystemAssetStore;
+
+    /**
+     * @var PublicAssetAdapter
+     */
+    private $publicAssetAdapter;
+
     /**
      * @param array $options
      * @return array An array containing results
      * @throws \SilverStripe\ORM\ValidationException
      */
     public function run(array $options)
+    {
+        $this->prepare($options);
+
+        // Check for db files with missing filesystem files and remove those db entries
+        $this->deleteBroken();
+
+        // Sync Files
+        $this->syncFiles();
+
+        // Remove Directories
+        $this->removeDirectories();
+
+        return $this->results;
+    }
+
+    /**
+     * Prepare for execution
+     * @param $options
+     * @throws ValidationException
+     */
+    private function prepare($options)
     {
         // Set max time and memory limit
         Environment::increaseTimeLimitTo();
         Environment::increaseMemoryLimitTo();
 
         // Prepare for execution
         $this->allowedExtensions = File::getAllowedExtensions();
-        $this->path = isset($options['path']) ? $options['path'] : ASSETS_PATH;
```

Comment:
Could this break someone's setup (who's not running on a managed platform with Flysystem)? Maybe we can't make this change in a minor version...

Reply:
This task requires Flysystem. There is an exception further down that gets thrown if it can't find it.